src/modules/cpu/kernel/glitch.hpp

```diff
@@ -0,0 +1,648 @@
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+inline void compute_src_loc(int row , int col, Rpp32s *locArray, RpptDescPtr srcDescPtr, RpptChannelOffsets *rgbOffsets, RpptROI roi, int batchCount, int channelValue)
+{
+    int xR, yR, xG, yG, xB, yB;
+
+    xR = col + rgbOffsets[batchCount].r.x;
+    yR = row + rgbOffsets[batchCount].r.y;
+
+    xG = col + rgbOffsets[batchCount].g.x;
+    yG = row + rgbOffsets[batchCount].g.y;
+
+    xB = col + rgbOffsets[batchCount].b.x;
+    yB = row + rgbOffsets[batchCount].b.y;
+
+    if (xR >= roi.xywhROI.roiWidth || xR < roi.xywhROI.xy.x || yR >= roi.xywhROI.roiHeight || yR < roi.xywhROI.xy.y)
+    {
+        xR = col;
+        yR = row;
+    }
+
+    if (xG >= roi.xywhROI.roiWidth || xG < roi.xywhROI.xy.x || yG >= roi.xywhROI.roiHeight || yG < roi.xywhROI.xy.y)
+    {
+        xG = col;
+        yG = row;
+    }
+
+    if (xB >= roi.xywhROI.roiWidth || xB < roi.xywhROI.xy.x || yB >= roi.xywhROI.roiHeight || yB < roi.xywhROI.xy.y)
+    {
+        xB = col;
+        yB = row;
+    }
+
+    locArray[0] = yR * srcDescPtr->strides.hStride + xR * channelValue;
+    locArray[1] = yG * srcDescPtr->strides.hStride + xG * channelValue;
+    locArray[2] = yB * srcDescPtr->strides.hStride + xB * channelValue;
+}
+
+RppStatus glitch_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                   RpptDescPtr srcDescPtr,
+                                   Rpp8u *dstPtr,
+                                   RpptDescPtr dstDescPtr,
+                                   RpptChannelOffsets *rgbOffsets,
+                                   RpptROIPtr roiTensorPtrSrc,
+                                   RpptRoiType roiType,
+                                   RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
```

Comment:
@HazarathKumarM 
Please get the numThreads from handle instead using batchsize

Please ensure to follow this approach of getting numThreads from handle and use it for new CPU kernels henceforth
Reference added below
https://github.com/ROCm/rpp/blob/master/src/modules/cpu/kernel/brightness.hpp#L41

It needs to be done for all bit depth variants in HOST

Reply:
Done